File: src/test/java/no/nav/aareg/teknisk/historikk/ActuatorTest.java
package no.nav.aareg.teknisk.historikk;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.web.servlet.client.RestTestClient;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ActuatorTest extends AaregTekniskHistorikkTest {

    @Autowired
    private RestTestClient restTestClient;

    @Test
    void healthEndepunktErTilgjengelig() {
        restTestClient.get().uri("/actuator/health").exchange()
                .expectStatus().isOk()
                .expectBody()
                .jsonPath("status").isEqualTo("UP")
                .jsonPath("groups").isArray()
                .jsonPath("groups").value(groups -> assertEquals(2, ((List<?>) groups).size()))
                .jsonPath("groups").value(groups -> assertEquals("liveness", ((List<?>) groups).get(0)))
                .jsonPath("groups").value(groups -> assertEquals("readiness", ((List<?>) groups).get(1)));
    }

    @Test
    void livenessProbeErTilgjengelig() {
        restTestClient.get().uri("/actuator/health/liveness").exchange()
                .expectStatus().isOk();
    }

    @Test
    void readinessProbeErTilgjengelig() {
        restTestClient.get().uri("/actuator/health/readiness").exchange()
                .expectStatus().isOk();
    }

    @Test
    void prometheusMetrikkerErTilgjengelig() {
        restTestClient.get().uri("/actuator/prometheus").exchange()
                .expectStatus().isOk();
    }
}

File: src/test/kotlin/no/nav/aareg/teknisk_historikk/ActuatorTest.kt
package no.nav.aareg.teknisk_historikk

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.http.HttpStatus

class ActuatorTest: AaregTekniskHistorikkTest() {
    @Autowired
    lateinit var testRestTemplate: TestRestTemplate

    @Test
    fun `health-endepunkt er tilgjengelig`(){
        val resultat = testRestTemplate.getForEntity("/actuator/health", String::class.java)

        assertEquals(HttpStatus.OK, resultat.statusCode)
        assertEquals("""
            {"status":"UP","groups":["liveness","readiness"]}
        """.trimIndent(), resultat.body)
    }

    @Test
    fun `liveness-probe er tilgjengelig`(){
        val resultat = testRestTemplate.getForEntity("/actuator/health/liveness", String::class.java)

        assertEquals(HttpStatus.OK, resultat.statusCode)
    }

    @Test
    fun `readiness-probe er tilgjengelig`(){
        val resultat = testRestTemplate.getForEntity("/actuator/health/readiness", String::class.java)

        assertEquals(HttpStatus.OK, resultat.statusCode)
    }

    @Test
    fun `prometheus-metrikker er tilgjengelig`(){
        val resultat = testRestTemplate.getForEntity("/actuator/prometheus", String::class.java)

        assertEquals(HttpStatus.OK, resultat.statusCode)
    }
}